core/src/main/scala/org/bitcoins/core/util/BitcoinScriptUtil.scala

```diff
@@ -671,6 +721,103 @@ trait BitcoinScriptUtil extends BitcoinSLogger {
     }
     ScriptInterpreter.runAllVerify(programs)
   }
+
+  def verifyPSBTInputScript(
```

Comment:
can you not use the TxSigComponent method that was added for any of this?

Reply:
Possibly but will require a bunch of work to get a valid input info from a signed tx + input map. Maybe better for a separate PR.

Reply:
I'm with Nadav, it's pretty hard to read this. It would be useful to define helper methods to delegate to and also be more specific types in the parameters so it doesn't have to be monolithic. Simple helper methods like

```
  private def verifyWitnessPSBTInputScript(
      wtx: WitnessTransaction,
      inputPSBTMap: InputPSBTMap,
      index: Int,
      flags: Seq[ScriptFlag] = Policy.standardFlags): Try[Transaction] = {
      //..handle witness inputs here
  }
```

go a long way for improving readability. If you want to address it on a separate PR i'm fine with that, but file an issue please